completions/zsh/_brew

```diff
@@ -3,154 +3,825 @@
+    'on[turn on analytics]' \
+    'regenerate-uuid[regenerate UUID used in analytics]' \
+    'state[display anonymous user behaviour analytics state]'
+}
+
+# brew aspell_dictionaries
+_brew_aspell_dictionaries() {
+  return 1
+}
+
+# brew audit [--strict] [--online] [--new-formula] [--display-cop-names] [--display-filename] [formulae]
+_brew_audit() {
+  _arguments \
+    '(--new-formula)--strict[run additional checks including RuboCop style checks]' \
+    '(--new-formula)--online[run slower checks that require a network connection]' \
+    '(--online --strict)--new-formula[check if a new formula is eligable for Homebrew. Implies --strict and --online]' \
+    '--display-cop-names[include RuboCop cop name for each violation in the output]' \
+    '--display-filename[prefix every line of output with the name of the file or formula being audited]' \
+    '*:formula:__brew_formulae'
+}
+
+# brew boneyard-formula-pr [--dry-run] [--local]  formula-name
+_brew_boneyard_formula_pr() {
+  _arguments \
+    '--dry-run[print what would be done rather than doing it]' \
+    '--local[perform only local operations(i.e. don''t push or create PR)]' \
+    ':formula:__brew_formulae'
+}
+
+# brew bottle [--verbose] [--no-rebuild] [--keep-old] [--skip-relocation] [--root-url=root_url]
+# brew bottle --merge [--no-commit] [--keep-old] [--write]
+# TODO missing argument docs
+_brew_bottle() {
+  _arguments \
+    '--keep-old' \
+    - set1 \
+      '--verbose' \
+      '--no-rebuild' \
+      '--skip-relocation' \
+      '--root-url=' \
+    - set2 \
+      '--merge' \
+      '--no-commit' \
+      '--write'
+}
+
+# brew bump-formula-pr [--devel] [--dry-run] [--audit|--strict] --url=url --sha256=sha-256 formula
+# brew bump-formula-pr [--devel] [--dry-run] [--audit|--strict] --tag=tag --revision=revision formula
+_brew_bump_formula_pr() {
+  _arguments \
+      '--devel[bump the development rather than stable version]' \
+      '--dry-run[print what would be done rather than doing it]' \
+      '(--strict)--audit[run brew audit before opening the PR]' \
+      '(--audit)--strictrun brew audit --strict before opening the PR]' \
+    - set1 \
+      '--url=-[new url]:url:_urls' \
+      '--sha256=-[new sha256 checksum]:sha256: ' \
+      ': :__brew_formulae' \
+    - set2 \
+      '--tag=-[new tag]: ' \
+      '--revision=-[new revision]: ' \
+      ': :__brew_formulae'
+}
+
+# brew cat formula
+_brew_cat() {
+  _arguments \
+    '1:: :__brew_formulae'
+}
+
+# brew cleanup [--prune=days] [--dry-run] [-s] [formulae]
+_brew_cleanup() {
+  _arguments \
+    '--prune=-[remove all cache files older than days]:days: ' \
+    '(--dry-run,-n)'{--dry-run,-n}'[show what would be removed, but do not actually remove anything]' \
+    '-s[scrub cache]' \
+    '*::formula:__brew_formulae'
+}
+
+# brew command cmd
+_brew_command() {
+  _arguments \
+    ': :__brew_all_commands'
+}
+
+# brew commands [--quiet [--include-aliases]]
+_brew_commands() {
```

Comment:
I'm not sure the logic here is ideal. It seems to have the effect of not showing `--include-aliases` in the list of options if you just try to complete `brew commands --`. But if you complete `brew commands --i`, it'll complete to `--include-aliases`, regardless of whether `--quiet` is already specified. And if `--quiet` is specified, trying to complete anything fills in `--include-aliases`. Is that intentional? The comment above seems to indicate that `--include-aliases` is a sub-option of `--quiet`, and that matches my reading of the source for `brew commands`.

Strictly speaking, this might be the right behavior, but it feels a little odd from a usability perspective. I think it might be better to just present both `--quiet` and `--include-aliases` as options, along with a description for each, even though there's a dependency there.

Concrete example: if you do `brew commands ` and hit tab, it first fills in `--quiet`, and if you hit tab again, it fills in `--include-aliases`. This feels a little odd to me, even though it might be technically correct. I feel like as a user, I'd rather see a menu with both `--quiet` and `--include-aliases` and a description of each, instead of them each popping up in turn.